Assets/BossRoom/Scripts/Client/Net/ClientGameNetPortal.cs

```diff
@@ -211,6 +218,91 @@ public static bool StartClientRelayMode(GameNetPortal portal, string roomKey, ou
             return true;
         }
 
+        public static bool StartClientUnityRelayMode(GameNetPortal portal, string joinCode, out string failMessage)
+        {
+
+            var chosenTransport  = NetworkManager.Singleton.gameObject.GetComponent<TransportPicker>().UnityRelayTransport;
+            NetworkManager.Singleton.NetworkConfig.NetworkTransport = chosenTransport;
+
+            switch (chosenTransport)
+            {
+                case UnityTransport utp:
+                    Debug.Log($"Setting Unity Relay client with join code {joinCode}");
+                    Task t = Task.Run( () => {UnityServices.InitializeAsync();});
+                    if (!t.Wait(30000))
+                    {
+                        Debug.Log("UnityServices.Initialize did not finish within 30 seconds");
+                    }
+                    Debug.Log(AuthenticationService.Instance);
+                    if (!AuthenticationService.Instance.IsSignedIn)
+                    {
+                        Task t2 = Task.Run( () => {AuthenticationService.Instance.SignInAnonymouslyAsync();});
+                        t2.Wait(30000);
+                        var playerId = AuthenticationService.Instance.PlayerId;
+                        Debug.Log(playerId);
+
+                        var clientRelayUtilityTask = RelayUtility.JoinRelayServerFromJoinCode(joinCode);
+                        clientRelayUtilityTask.Wait(30000);
+                        if (clientRelayUtilityTask.IsFaulted)
+                        {
+                            throw new Exception($"Failed to allocate on relay server {clientRelayUtilityTask.Exception?.Message}");
+                        }
+
+                        var (ipv4address, port, allocationIdBytes, connectionData, hostConnectionData, key) = clientRelayUtilityTask.Result;
+
+                        utp.SetRelayServerData(ipv4address, port, allocationIdBytes, key, connectionData, hostConnectionData);

```

Comment:
It seems weird that this is user side code, that seems like the kind of thing that should be automated for us

Reply:
This is no different than setting an IP Address for a Transport? As that is the connection data required you set it and then we use it. This is the only entry point for the Relay in the adapter/transport

Reply:
I was saying this since we already ask to join earlier with JoinRelayServerFromJoinCode, it seemed like it should have been just one call from a user's perspective.
But I'm guessing if the goal is to have relay be usable standalone, then having that dependency with UTP would break this?
If that's the case, I'll resolve that comment